test/OpenTelemetry.Instrumentation.AWSLambda.Tests/Implementation/AWSMessagingUtilsTests.cs

```diff
@@ -106,11 +106,11 @@ public void ExtractParentContext_SetParentFromMessageBatchIsEnabled_ParentIsSetF
             },
         };
 
-        (PropagationContext parentContext, IEnumerable<ActivityLink> links) = AWSMessagingUtils.ExtractParentContext(sqsEvent);
+        (PropagationContext parentContext, IEnumerable<ActivityLink>? links) = AWSMessagingUtils.ExtractParentContext(sqsEvent);
 
         Assert.NotEqual(default, parentContext);
         Assert.Equal(SpanId1, parentContext.ActivityContext.SpanId.ToHexString());
-        Assert.Single(links);
+        Assert.Single(links ?? Array.Empty<ActivityLink>());
```

Comment:
```suggestion
        Assert.Single(links!);
```

Adding null-safety in unit tests is IMHO counter-productive. If you don't expect a null, you should not try to insert some fallback values, making the error harder to understand.